tfhe/src/core_crypto/entities/modulus_switched_lwe_ciphertext.rs

```diff
@@ -0,0 +1,143 @@
+use super::lwe_ciphertext::{LweCiphertext, LweCiphertextOwned};
+use crate::core_crypto::commons::ciphertext_modulus::CiphertextModulus;
+use crate::core_crypto::prelude::*;
+
+pub struct PackedModulusSwitchedLweCiphertext {
+    packed_coeffs: Vec<u64>,
+    lwe_dimension: LweDimension,
+    log_modulus: u64,
+    ciphertext_modulus: CiphertextModulus<u64>,
+}
+
+impl PackedModulusSwitchedLweCiphertext {
+    pub fn compress(ct: &LweCiphertext<&[u64]>, log_modulus: u64) -> Self {
+        let decomposer = SignedDecomposer::new(
+            DecompositionBaseLog(log_modulus as usize),
+            DecompositionLevelCount(1),
+        );
```

Comment:
I'm wondering if we should use the fast_pbs_modulus_switch instead ? to be sure formulas agree

Reply:
`fast_pbs_modulus_switch` can return `2^log_modulus` so i don't think we can use it as is here

Reply:
I remember that, but that would be equivalent to a 0 right ?

Reply:
yep, which can be from 600 to 1000 + coeffs

Reply:
I read "hundreds of thousands of times"
A few instructions would be negligible
We could change this behavior IMO

Reply:
Changed